core/src/main/java/com/linecorp/armeria/internal/IdleTimeoutHandler.java

```diff
@@ -32,14 +32,24 @@
     private static final Logger logger = LoggerFactory.getLogger(IdleTimeoutHandler.class);
 
     private final String name;
+    private boolean isHttp2;
 
-    protected IdleTimeoutHandler(String name, long idleTimeoutMillis) {
+    protected IdleTimeoutHandler(String name, long idleTimeoutMillis, boolean isHttp2) {
         super(0, 0, idleTimeoutMillis, TimeUnit.MILLISECONDS);
         this.name = requireNonNull(name, "name");
+        this.isHttp2 = isHttp2;
     }
 
+    /**
+     * If the channel is serving http2 the we will leverage sending PING's using {@link Http2KeepAliveHandler}
+     * but if it is http1.1 channel then we will close the channel
+     */
     @Override
     protected final void channelIdle(ChannelHandlerContext ctx, IdleStateEvent evt) throws Exception {
+        if (isHttp2) {
```

Comment:
I had 2 options:

1. I had to use isHttp2 flag to indicate that connection is http2
2. Or I could remove `IdleTimeoutHandler` and add `IdleStateHandler` (netty's base implementation).

I choose (1), please let me know if there are better alternatives.

Reply:
+1 for using a flag. :smile:

Reply:
(Because it's better keeping the length of a Netty pipeline for performance's sake)